### ISSUE TITLE
Create web user returns server error

### ISSUE THREAD

**Issue by florianm** [opendatakit/central#81]
# Setup
[ODK Central 0.5 on rancher/K8s/Azure](https://forum.opendatakit.org/t/installing-odk-central-in-microsoft-azure-cloud/18020/2).

Our config contains our own mail server with settings that are confirmed working in other applications running out of the same K8s cluster.

# Issue
* Logged in to ODK Central
* Users > Create Web User > enter email and name > Create
* Error message "Something went wrong: the server returned an invalid error." 
* Error log on server:
```
IP - - [27/Jun/2019:03:59:20 +0000] "POST /v1/users HTTP/1.1" 500 348 "https://odkcentral.dbca.wa.gov.au/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0"
```

# Related
* Creating a web user through CLI as per [docs](https://docs.opendatakit.org/central-command-line/#creating-a-web-user-by-command-line) works - so I assume the backend has at least one working pathway to create users. This step doesn't involve sending an email invite.
* Editing the user profile through the Central frontend works. This step doesn't involve sending an email invite.

Which steps could I take next to narrow down the problem?

**Comment by issa-tseng** [comment]
so you are probably correct in identifying the mail server configuration as the problem. the CLI path does not send an email.

the next step is probably to dump the `service` container logs rather than the `nginx` logs. there should be a stack trace in there somewhere.

**Comment by florianm** [comment]
Thought so too! 

Another try: tried resetting my password which should have sent an email. "Something went wrong". It's the mail server for sure.

However, there's nothing in the service container's logs. Is there a way to turn up their verbosity?

**Comment by issa-tseng** [comment]
ah right i forgot: because we run a multi-process cluster for the service container, we don't log out as usual. i should see if that can be corrected.

in the meantime, the error log for service will be at `/var/log/odk/stderr.log`.

**Comment by florianm** [comment]
Cheers - I can see stdout.log, e.g. on "reset password" and resulting ISE:
```
::ffff:10.42.1.239 - - [28/Jun/2019:01:15:20 +0000] "POST /v1/users/reset/initiate HTTP/1.0" 500 348
```

and stderr.log:
```
root@service-HASH:/var/log/odk# tail -n 200 -f stderr.log 
{ Error: self signed certificate
    at TLSSocket.<anonymous> (_tls_wrap.js:1105:38)
    at emitNone (events.js:106:13)
    at TLSSocket.emit (events.js:208:7)
    at TLSSocket._finishInit (_tls_wrap.js:639:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:469:38) code: 'ESOCKET', command: 'CONN' }tail: unrecognized file system type 0x794c7630 for 'stderr.log'. please report this to bug-coreutils@gnu.org. reverting to polling
```

I will check our SSL certificate. That means ODK Central works as expected, although I needed that reminder to check /var/logs. I'm closing this issue.

I'd love to have those logs in an easily accessible place - streaming to the container's logs would be just fine. Also, see the error about unrecognised file system type of the log file (`tail` throws that on both stderr and stdout).

**Comment by florianm** [comment]
ps setting up a project in Sentry.io and placing the Sentry config in local.json (take defaults.json as template) makes the stderr logs report nicely to Sentry.io. That means logging is fine, maybe a note in the install docs on setting up one's own logging (next to the note on how to disable ODK's own Sentry logging) would be helpful.

**Comment by issa-tseng** [comment]
i'm honestly pretty surprised `tail` works at all; the docker distributions tend to be ridiculously stripped down.

i'll add the note about adding logging when i get a chance; thanks!

**Comment by florianm** [comment]
Fixed the email problem in #105
